lib/utils/aws/internal/region/codegen.go

```diff
@@ -0,0 +1,160 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"text/template"
+
+	"github.com/gravitational/trace"
+)
+
+var tmpl = template.Must(template.New("generate").
+	Funcs(map[string]any{
+		"quote": func(v string) string {
+			return strconv.Quote(v)
+		},
+	}).
+	Parse(`
+// Code generated by internal/region/codegen.go. DO NOT EDIT.
+
+package aws
+
+// Region represents an AWS region.
+type Region struct {
+	// Name is the region name.
+	Name string
+	// Description is the human readable region name.
+	Description string
+}
+
+// regions is the set of available AWS regions based on the AWS SDK definition.
+var regions = map[string]Region{
+	{{- range $name, $desc := $.Regions }}
+	{{ quote $name }}: Region{
+		Name: {{ quote $name }},
+		Description: {{ quote $desc }},
+	},
+	{{- end }}
+}
+`))
+
+// TemplateData is the data passed to the template.
+type TemplateData struct {
+	Regions map[string]string
+}
+
+// AWSPartitions contains the struct representation of the AWS SDK patitions
+// JSON file.
+type AWSPartitions struct {
+	Partitions []struct {
+		Regions map[string]struct {
+			Description string `json:"description"`
+		} `json:"regions"`
+	} `json:"partitions"`
+}
+
+func main() {
+	partitionsContents, err := loadSDKPartitionsFile()
+	if err != nil {
+		log.Fatalf("failed to parse patitions: %s", trace.DebugReport(err))
+	}
+
+	data, err := convertToTemplateData(partitionsContents)
+	if err != nil {
+		log.Fatalf("failed to parse partitions file: %s", trace.DebugReport(err))
+	}
+
+	file, err := os.Create(outputFilename)
+	if err != nil {
+		log.Fatalf("failed to create output file: %v", err)
+	}
+	defer func() {
+		if err := file.Close(); err != nil {
+			log.Fatalf("failed to close output file: %v", err)
+		}
+	}()
+
+	if err := tmpl.Execute(file, data); err != nil {
+		log.Fatalf("failed to generate file: %v", err)
+	}
+}
+
+// ListModule is the struct representation of the `go list` module output.
+// Only the necessary fields are included.
+type ListModule struct {
+	// Version is module version
+	Version string
+	// Dir is directory holding local copy of files, if any
+	Dir string
+}
+
+func convertToTemplateData(partitionsContents []byte) (*TemplateData, error) {
+	var awsPartitions AWSPartitions
+	if err := json.Unmarshal(partitionsContents, &awsPartitions); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	data := &TemplateData{
+		Regions: make(map[string]string),
+	}
+
+	for _, partition := range awsPartitions.Partitions {
+		for name, region := range partition.Regions {
+			data.Regions[name] = region.Description
+		}
+	}
+
+	return data, nil
```

Comment:
i would double-check that the list is non-empty here in case the format changed.

Reply:
Added a check on the parsed JSON struct.